File: SameTree/third.py
The "proper" way:

class Solution:
    def isSameTree(self, p, q):
        if p and q:
            return p.val == q.val and self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)
        return p == q
The "tupleify" way:

class Solution:
    def isSameTree(self, p, q):
        def t(n):
            return n and (n.val, t(n.left), t(n.right))
        return t(p) == t(q)

File: MaximumDepthofBinaryTree/third.java
public class Solution {
public int maxDepth(TreeNode root) {
    Stack<TreeNode> stack=new Stack<TreeNode>();
    Stack<Integer> values=new Stack<Integer>();
    if(root==null)return 0;
    stack.push(root);
    int max=0;
    values.push(1);
    while(!stack.isEmpty()){
        TreeNode node=stack.pop();
        int value=values.pop();
        max=Math.max(value,max);
        if(node.right!=null){
            stack.push(node.right);
            values.push(value+1);
        }
        if(node.left!=null){
            stack.push(node.left);
            values.push(value+1);
        }
    }
    return max;
}
}
